packages/jobs/gateways/contractProxy.js

```diff
@@ -66,16 +64,45 @@ function createProxy(contractName) {
         abi = CurrentContract.abi;
       }
 
-      let web3Contract = web3.eth.contract(abi).at(address);
-
-      // TODO move to web3 1.0 promievents and drop Bluebird.
-      // TODO allow contract proxies to redefine contract methods
-      //      for different contract versions.
-      return Promise.promisifyAll(web3Contract);
+      return new ethers.Contract(address, abi, wallet);
     },
   };
 }
 
+async function getAllContractsForDocument(project, wallet) {
```

Comment:
It's not clear for me, are we connecting a wallet or a signer here?

Reply:
Wallet implements the Signer API